protocol/ticket.re

```diff
@@ -9,7 +9,16 @@ type t =
 
 // TODO: should we expose to_yojson here?
 let compare = (a, b) => {
-  let to_string = a =>
-    Bytes.to_string(a.data) ++ Tezos_interop.Address.to_string(b.ticketer);
```

Comment:
This is exactly why I hate functions with two idents with same type.

Reply:
I think this is a variable naming issue. Naming is always tricky and our language thankfully provides ways to avoid naming altogether. Also, we could avoid declaring pure functions in local scope.

Reply:
Yes and avoiding naming makes it really hard to reason, point free style is mostly a regression from readability. 

> Also, we could avoid declaring pure functions in local scope.

Not so sure I agree here, this function is an implementation detail, it should not pollute the scope for everyone else.